cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/query/ForwardingDatastore.java

```diff
@@ -0,0 +1,181 @@
+package com.cloudant.sync.query;
+
+import com.cloudant.sync.datastore.Changes;
+import com.cloudant.sync.datastore.ConflictException;
+import com.cloudant.sync.datastore.ConflictResolver;
+import com.cloudant.sync.datastore.Datastore;
+import com.cloudant.sync.datastore.DocumentException;
+import com.cloudant.sync.datastore.DocumentNotFoundException;
+import com.cloudant.sync.datastore.DocumentRevision;
+import com.cloudant.sync.event.EventBus;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Created by rhys on 04/07/2016.
+ */
+public class ForwardingDatastore implements Datastore {
```

Comment:
What is the point of this class, I can't see any method interception going on here only pure delegation?
If we need something that exposes the `Datastore` interface only but not the impl, why not just do
`Datastore fd = ds;`?
Also if we do need this it would be much simpler to implement with a `java.lang.reflect.Proxy` and an `InvocationHandler`.